File: mdoc/src/main/scala/mdoc/internal/worksheets/ImportedScriptFile.scala
package mdoc.internal.worksheets

import java.{util => ju}
import mdoc.{interfaces => i}
import java.nio.file.Path

final case class ImportedScriptFile(
    val path: Path,
    val packageName: String,
    val objectName: String,
    val instrumentedSource: String,
    val originalSource: String,
    val files: ju.List[i.ImportedScriptFile]
) extends i.ImportedScriptFile

File: mdoc-interfaces/src/main/scala/mdoc/interfaces/ImportedScriptFile.java
package mdoc.interfaces;

import java.util.List;
import java.nio.file.Path;

public abstract class ImportedScriptFile {

  public abstract Path path();
  public abstract String packageName();
  public abstract String objectName();
  public abstract String instrumentedSource();
  public abstract String originalSource();
  public abstract List<ImportedScriptFile> files();

}
